slides/week3/git.org

```diff
@@ -25,12 +28,23 @@
 - Branches are what people have the most trouble with
   - _*Ask questions /please/!*_ This is a difficult topic.
 
-** Git As a Graph
+** Remotes vs Branches vs Forks
+- A fork or clone does a full copy of the git repo
+- A remote is a pointer to another copy of this git repo
+- A repository is a collection of *branches*
+- You can only directly modify local branches, but:
+  - You can merge from a remote into your branch (pull)
+  - You can merge from your branch to a remote (push)
 
-- git history is a *graph*
+** Git as a Graph
+- Git history is a *graph*
+- A graph is simply a collection of nodes connected by edges
 - Each commit is a node on the graph
-- Ways to see it as a graph, rather than a line of commits
-  - ~git~ porcelain
+- Subsequent commits are linked by edges
+
+** Git as a Graph cont.
+- Some useful tools to visualize your repo's git history as a graph:
+  - ~git porcelain~
     #+BEGIN_SRC sh
       git log --oneline --graph --all --decorate
```

Comment:
might be nice to show this on the software training repo since ours branches a lot.

Reply:
Isn't that what the output on this slide is showing?

Reply:
Valid point